components/public-api-server/middleware/logging_test.go

```diff
@@ -0,0 +1,39 @@
+package middleware
+
+import (
+	"bytes"
+	log "github.com/sirupsen/logrus"
+	_ "github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestLoggingMiddleware(t *testing.T) {
+	logInMemory := &bytes.Buffer{}
+	logger := log.New()
+	logger.SetOutput(logInMemory)
+
+	expectedBody := `hello world`
+
+	someHandler := http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.Write([]byte(expectedBody))
+	})
+	req := httptest.NewRequest("GET", "/", nil)
+	rec := httptest.NewRecorder() // this records the response
+
+	m := NewLoggingMiddleware(logger)
```

Comment:
@easyCZ I'm having trouble injecting the logger here. It's expecting type `*logrus.Entry` but I couldn't figure out how to pass that in 🙏🏽